kura/test/org.eclipse.kura.nm.test/src/test/java/org/eclipse/kura/nm/NMDbusConnectorTest.java

```diff
@@ -0,0 +1,575 @@
+    }
+    
+    @Test
+    public void getInterfaceStatusShouldWorkUnsuported() throws DBusException {
+        givenMockedDevice("unused0", NMDeviceType.NM_DEVICE_TYPE_UNUSED1, NMDeviceState.NM_DEVICE_STATE_FAILED);
+        givenExtraStatusMocksFor("unused0", NMDeviceState.NM_DEVICE_STATE_FAILED);
+        givenMockedDeviceList();
+        givenNetworkServiceThatAlwaysReturnsEmpty();
+        
+        whenGetInterfaceStatus("unused0", networkService);
+        
+        thenNoExceptionIsThrown();
+        thenInterfaceStatusIsNull();
+    }
+    
+    @Test
+    public void getInterfaceStatusShouldWorkWireless() throws DBusException {
+        givenMockedDevice("wlan0", NMDeviceType.NM_DEVICE_TYPE_WIFI, NMDeviceState.NM_DEVICE_STATE_FAILED);
+        givenExtraStatusMocksFor("wlan0", NMDeviceState.NM_DEVICE_STATE_FAILED);
+        givenMockedDeviceList();
+        givenNetworkServiceThatAlwaysReturnsEmpty();
+        
+        whenGetInterfaceStatus("wlan0", networkService);
+        
+        thenNoExceptionIsThrown();
+        thenInterfaceStatusIsNull();
+    }
+    
+    @Test
+    public void getInterfaceStatusShouldWorkEthernetUSB() throws DBusException {
+        givenMockedDevice("eth0", NMDeviceType.NM_DEVICE_TYPE_ETHERNET, NMDeviceState.NM_DEVICE_STATE_ACTIVATED);
+        givenExtraStatusMocksFor("eth0", NMDeviceState.NM_DEVICE_STATE_ACTIVATED);
+        givenMockedDeviceList();
+        givenNetworkServiceMockedForUsbInterfaceWithName("eth0");
+        
+
+        whenGetInterfaceStatus("eth0", networkService);
+
+        thenNoExceptionIsThrown();
+        thenInterfaceStatusIsNotNull();
+        thenNetInterfaceTypeIs(NetworkInterfaceType.ETHERNET);
+    }
+
+    public void givenBasicMockedDbusConnector() throws DBusException {
+		when(dbusConnection.getRemoteObject(eq("org.freedesktop.NetworkManager"), eq("/org/freedesktop/NetworkManager"), any()))
+					.thenReturn(mockedNetworkManager);
+		
+		this.instanceNMDbusConnector = NMDbusConnector.getInstance(this.dbusConnection);
+	}
+
+    public void givenMockedPermissions() {
+
+        Map<String, String> tempPerms = new HashMap<>();
+
+        tempPerms.put("test1", "testVal1");
+        tempPerms.put("test2", "testVal2");
+        tempPerms.put("test3", "testVal3");
+
+        when(mockedNetworkManager.GetPermissions()).thenReturn(tempPerms);
+
+    }
+
+    public void givenMockedVersion() throws DBusException {
+
+        Properties mockProps = mock(org.freedesktop.dbus.interfaces.Properties.class);
+        when(mockProps.Get(eq("org.freedesktop.NetworkManager"), eq("Version"))).thenReturn("Mock-Version");
+
+        doReturn(mockProps).when(this.dbusConnection).getRemoteObject(eq("org.freedesktop.NetworkManager"),
+                eq("/org/freedesktop/NetworkManager"), eq(Properties.class));
+    }
+    
+    public void givenMockedDevice(String interfaceName, NMDeviceType type, NMDeviceState state) throws DBusException {
```

Comment:
For easier testing of devices without an associated connection I would suggest you to add a parameter here:

```suggestion
    public void givenMockedDevice(String interfaceName, NMDeviceType type, NMDeviceState state, boolean hasAssociatedConnection) throws DBusException {
```